mojave/code/modules/preferences/character.dm

```diff
@@ -0,0 +1,185 @@
+	should_generate_icons = TRUE
+
+/datum/preference/choiced/undershirt/init_possible_values()
+	var/icon/body = icon('icons/mob/human_parts_greyscale.dmi', "human_r_leg")
+	body.Blend(icon('icons/mob/human_parts_greyscale.dmi', "human_l_leg"), ICON_OVERLAY)
+	body.Blend(icon('icons/mob/human_parts_greyscale.dmi', "human_r_arm"), ICON_OVERLAY)
+	body.Blend(icon('icons/mob/human_parts_greyscale.dmi', "human_l_arm"), ICON_OVERLAY)
+	body.Blend(icon('icons/mob/human_parts_greyscale.dmi', "human_r_hand"), ICON_OVERLAY)
+	body.Blend(icon('icons/mob/human_parts_greyscale.dmi', "human_l_hand"), ICON_OVERLAY)
+	body.Blend(icon('icons/mob/human_parts_greyscale.dmi', "human_chest_m"), ICON_OVERLAY)
+	body.Blend("#ffe0d1", ICON_MULTIPLY)
+
+	var/list/values = list()
+
+	for (var/accessory_name in GLOB.undershirt_list)
+		var/icon/icon_with_undershirt = icon(body)
+
+		if (accessory_name != "Nude")
+			var/datum/sprite_accessory/accessory = GLOB.undershirt_list[accessory_name]
+			icon_with_undershirt.Blend(icon('mojave/icons/mob/clothing/underwear.dmi', accessory.icon_state), ICON_OVERLAY) //MOJAVE SUN EDIT - Prefs
+
+		icon_with_undershirt.Crop(9, 9, 23, 23)
+		icon_with_undershirt.Scale(32, 32)
+		values[accessory_name] = icon_with_undershirt
+
+	return values
+
+/datum/preference/choiced/undershirt/apply_to_human(mob/living/carbon/human/target, value)
+	target.undershirt = value
+
+//Underwear
+
+/datum/preference/choiced/underwear
+	savefile_key = "underwear"
+	savefile_identifier = PREFERENCE_CHARACTER
+	main_feature_name = "Underwear"
+	category = PREFERENCE_CATEGORY_CLOTHING
+	should_generate_icons = TRUE
+
+/datum/preference/choiced/underwear/init_possible_values()
+	return generate_values_for_underwear(GLOB.underwear_list, list("human_chest_m", "human_r_leg", "human_l_leg"), COLOR_ALMOST_BLACK)
+
+/datum/preference/choiced/underwear/apply_to_human(mob/living/carbon/human/target, value)
+	target.underwear = value
+
+/datum/preference/choiced/underwear/is_accessible(datum/preferences/preferences)
+	if (!..(preferences))
+		return FALSE
+
+	var/species_type = preferences.read_preference(/datum/preference/choiced/species)
+	var/datum/species/species = new species_type
+	return !(NO_UNDERWEAR in species.species_traits)
+
+/datum/preference/choiced/underwear/compile_constant_data()
+	var/list/data = ..()
+
+	return data
+
+//Socks
+
+/datum/preference/choiced/socks
+	savefile_key = "socks"
+	savefile_identifier = PREFERENCE_CHARACTER
+	main_feature_name = "Socks"
+	category = PREFERENCE_CATEGORY_CLOTHING
+	should_generate_icons = TRUE
+
+/datum/preference/choiced/socks/init_possible_values()
+	return generate_values_for_underwear(GLOB.socks_list, list("human_r_leg", "human_l_leg"))
+
+/datum/preference/choiced/socks/apply_to_human(mob/living/carbon/human/target, value)
+	target.socks = value
+
+//Backpack/Storage
+
+#define TPACK "Travelpack"
+#define MBACKPACK "Millitary Backpack"
+#define SATCHEL "Satchel"
+#define SACK "Sack"
```

Comment:
The storage sack, and generally having a belt slot storage item on spawn is not really something we're wanting to do right now. The item itself is fine to stay in but shouldn't be given to anyone on spawn for a variety of reasons, chiefly that it interferes with starting equipment for **a lot** of roles. But also that it has some _storage balance concerns_ that go against **THE VISION**